File: src/main/golang/binarysearch/Palindrome_Integer.go
package binarysearch

import "strconv"

func Palindrome_Integer(num int) bool {
	numStr := strconv.Itoa(num)
	midIdx := len(numStr) / 2
	for start, end := 0, len(numStr) - 1; start < midIdx; start++ {
		if numStr[start] != numStr[end] {
			return false
		}
		end--
	}
	return true
}

File: src/main/java/binarysearch/Palindrome_Integer.java
package binarysearch;

public class Palindrome_Integer {

    public boolean solve(int num) {
        String numStr = String.valueOf(num);
        int end = numStr.length() - 1;
        int midIdx = numStr.length() / 2;

        for (int start = 0; start < midIdx; start++, end--) {
            if (numStr.charAt(start) != numStr.charAt(end)) {
                return false;
            }
        }
        return true;
    }

}
